salt/transport/zeromq.py

```diff
@@ -380,8 +380,9 @@ def __del__(self):
     def connect(self):
         if not self.auth.authenticated:
             yield self.auth.authenticate()
-        self.publish_port = self.auth.creds['publish_port']
-        self._socket.connect(self.master_pub)
+        if self.opts.get('detect_mode', False) is False:
+            self.publish_port = self.auth.creds['publish_port']
+            self._socket.connect(self.master_pub)
```

Comment:
@DmitryKuzmenko this error is now showing up in the tests either way. Do you have any thoughts on where we can look?